lib/coralogix.js

```diff
@@ -216,44 +229,57 @@ const flush_stats = function coralogix_post_stats(ts, metrics) {
 
             payload.push({
                 labels: make_labels(mapping?.labels, host, get_prefix(key + '_sum')).concat(labels_from_tags),
-                samples: [{ timestamp: timestamp, value: sum }],
+                samples: [{timestamp: timestamp, value: sum}],
             });
 
             payload.push({
                 labels: make_labels(mapping?.labels, host, get_prefix(key + '_count')).concat(labels_from_tags),
-                samples: [{ timestamp: timestamp, value: count }],
+                samples: [{timestamp: timestamp, value: count}],
             });
 
 
             const buckets = mapping?.histogram_options?.buckets;
             if (buckets) {
                 const bucket_key = get_prefix(key) + '_bucket';
                 for (bucket in buckets) {
-                    const bucket_count = values.filter(function (v) { return v <= buckets[bucket]; }).length;
+                    const bucket_count = values.filter(function (v) {
+                        return v <= buckets[bucket];
+                    }).length;
                     const labels = make_labels(mapping?.labels, host, bucket_key).concat(labels_from_tags);
-                    labels.push({ name: 'le', value: String(buckets[bucket]) });
+                    labels.push({name: 'le', value: String(buckets[bucket])});
                     payload.push({
                         labels: labels,
-                        samples: [{ timestamp: timestamp, value: bucket_count }],
+                        samples: [{timestamp: timestamp, value: bucket_count}],
                     });
                 }
 
                 const labels = make_labels(mapping?.labels, host, bucket_key).concat(labels_from_tags);
-                labels.push({ name: 'le', value: '+Inf' });
+                labels.push({name: 'le', value: '+Inf'});
                 payload.push({
                     labels: labels,
-                    samples: [{ timestamp: timestamp, value: count }],
+                    samples: [{timestamp: timestamp, value: count}],
                 });
 
-            };
+            }
 
             // TODO support for quantiles
         }
     }
 
     post_stats(payload);
+    pruneTotalsAccumularot(ts);
 };
 
+function pruneTotalsAccumularot(currentTs){
+    totalsAccumulator.forEach( (value, key, _map) => {
+        let [_value,tsLiveUntil] = value;
+        if (currentTs >= tsLiveUntil){
+            console.log("Deleting key",key,(tsLiveUntil - currentTs))
```

Comment:
one log for every point - maybe too much?

Reply:
This should have been deleted.